backend/services/src/main/java/com/crimealert/services/AuthenticationService.java

```diff
@@ -0,0 +1,104 @@
+package com.crimealert.services;
+
+import org.bson.Document;
+
+import com.crimealert.models.Otp;
+import com.crimealert.models.User;
+import com.crimealert.utils.EmailSender;
+import com.crimealert.utils.SessionTokenUtils;
+import com.mongodb.client.MongoClient;
+import com.crimealert.Exceptions.ClientSideException;
+
+public class AuthenticationService {
+	
+	private UserService userService;
+	private DBConnectionService dbConnectionService;
+	private DBSearchService dbSearchService;
+	
+	
+	public  void requestOtp(String emailId)
+	{
+		MongoClient mongoClient = getDBConnectionService().getDBConnection();
+        
+        Document userRegistered = getDBSearchService().searchEmail(emailId, mongoClient);
+        
+        System.out.println("User Registered : " + userRegistered);
+        
+        if(userRegistered != null)
+        {
+		
+        	String token = SessionTokenUtils.generateToken(emailId);
+		
+        	String provider = getProvider(emailId);
+		
+        	if(EmailSender.sendOTP(emailId, token, provider))
+        	{
+        		Otp otp = new Otp();
+        		otp.setEmail(emailId);
+        		otp.setOneTimeToken(token);
+			
+        		getUserService().insertOtp(otp);
+        	}else
+        	{
+        		throw new ClientSideException("Failed to send otp");
+        	}
+        }
+        else
+        {
+        	throw new ClientSideException("User is not registered");
+        }
+		
+	}
+	
+	public void verifyOtp(Otp otp)
+	{
+		
+		Document foundOtp = getUserService().searchOtp(otp);
```

Comment:
Should we searchOtp with both otp and email? Just to prevent users from using others otp